src/dir.rs

```diff
@@ -244,6 +274,11 @@ impl Dir {
     /// respect to other threads and processes.
     ///
     /// Technically it means passing `O_EXCL` flag to open.
+    ///
+    /// Note that this method does not resolve symlinks by default, so you may have to call
```

Comment:
Will fail if path already created, even if it's symlink to a non-existing file.

Reply:
That makes sense; I just noticed the disclaimer in the docs. I think I will remove this comment, since it's irrelevant to this method.